src/dynamic_graph/dynamic-graph-py.cc

```diff
@@ -40,10 +46,22 @@ PyObject* plug(PyObject* /*self*/, PyObject* args) {
 
   pObjIn = PyCapsule_GetPointer(objIn, "dynamic_graph.Signal");
   SignalBase<int>* signalIn = (SignalBase<int>*)pObjIn;
-  if (signalIn == NULL) return NULL;
+  if (signalIn == NULL) {
+      struct module_state* st = GETSTATE(m);
+      std::ostringstream oss;
+      oss << "dgpy.plug in argument must be a dynamic_graph.Signal, not a " << PyCapsule_GetName(objIn);
```

Comment:
How about `dynamic_graph.plug(a, b): Argument 'a' must be of type 'dynamic_graph.Signal', but got 'xyz'`?

The string you propose is not wrong, but not everyone might understand what an "in argument" means here. Also, people usually do `import dynamic_graph` and then `dynamic_graph.plug`, so in that case, it's maybe not clear what `dgpy` means.

I try to design strings to be as much useful and understandable from the users perspective (and the knowledge/understanding the user is expected to have). 

Does that make sense?